drivers/source/serial_io.c

```diff
@@ -65,15 +73,29 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
 
-    /* USER CODE END */
+    if (xSemaphoreTake(mutex, UART_MUTEX_BLOCK_TIME) == pdFALSE)
+    {
+        return OBC_ERR_CODE_MUTEX_TAKE_FAILED;
+    }
+    else
+    {
+        sciSendBytes(text, length);
+        xSemaphoreGive(sciLinMutex);
+        return OBC_ERR_CODE_SUCCESS;
+    }
+
+    return OBC_ERR_CODE_MUTEX_TIMEOUT;
```

Comment:
This line is never reached because of the return in your else statement. In general, if you're returning inside your `if` statement, there's no need for an `else` statement.